### PR TITLE
Allow to separate bundle + rendering phase

### PR DESCRIPTION
**Pull request by JonnyBurger**

### CODE REVIEW

File: packages/cli/src/bundle.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+export const bundleCommand = async () => {
+	const file = parsedCli._[1];
+
+	if (!file) {
+		Log.error('No entry point specified. Pass more arguments:');
+		Log.error('   npx remotion bundle [entry-point]');
+		Log.error('Documentation: https://www.remotion.dev/docs/render');
+		process.exit(1);
+	}
+
```
Comment (JonnyBurger) [inline packages/cli/src/bundle.ts:16]: docs

File: packages/cli/src/parse-command-line.ts [hunk 2]
```diff
@@ -22,9 +22,11 @@ export type CommandLineOptions = {
 	['image-format']: ImageFormat;
 	['prores-profile']: ProResProfile;
 	['bundle-cache']: string;
+	['bundle-dir']: string;
 	['env-file']: string;
```
Comment (JonnyBurger) [inline packages/cli/src/parse-command-line.ts:26]: add docs for both

File: packages/core/src/config/bundle-out-dir.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+let bundleOutDir: string | null = null;
+
+export const getBundleOutDir = () => {
+	return bundleOutDir;
+};
+
+export const setBundleOutDir = (path: string) => {
+	bundleOutDir = path;
```
Comment (JonnyBurger) [inline packages/core/src/config/bundle-out-dir.ts]: Must be an absolute path, should do some validation

File: packages/core/src/config/index.ts [hunk 4]
```diff
@@ -61,6 +63,17 @@ export const Config = {
...
+		 * Define the "publicPath" used for the Webpack bundle.
+		 * This is the prefix of the URL pathname where the files in the public/ folder are put.
+		 * For example, if the bundle is hosted at the root of the host, it is `/`.
+		 * On Lambda, the bundle is usually hosted under `sites/{siteName}`
+		 */
+		setPublicPath,
+		/**
+		 * The path relative to the current working directory where the bundle gets saved.
+		 */
+		setBundleOutDir,
```
Comment (JonnyBurger) [inline packages/core/src/config/index.ts]: Document CLI

File: packages/core/src/config/public-path.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+let publicPath: string | null = null;
+
+export const getPublicPath = () => {
+	return publicPath;
+};
+
+export const setPublicPath = (path: string) => {
+	publicPath = path;
```
Comment (JonnyBurger) [inline packages/core/src/config/public-path.ts]: Should do some basic validation

File: packages/lambda/src/cli/commands/sites/create.ts [hunk 6]
```diff
@@ -33,19 +34,19 @@ export const sitesCreateSubcommand = async (args: string[]) => {
...
 		quit(1);
 	}
 
-	if (lstatSync(absoluteFile).isDirectory()) {
-		Log.error(
-			`You passed a path ${absoluteFile} but it is a directory. Pass a file instead.`
+	const needsToBundle = !(await stat(absoluteFile)).isDirectory();
+	if (!needsToBundle) {
+		Log.warn(
+			"Uploading a pre-bundled directory. Make sure it was bundled with the correct 'publicPath'."
```
Comment (JonnyBurger) [inline packages/lambda/src/cli/commands/sites/create.ts:50]: Document and add link here

**Comment by khanzzirfan** [comment]
Hey @JonnyBurger. Nothing related to the PR. but the commits messages something I think can be improved. 
wonder if you could follow the Commit-lint, and conventional commits. 
https://unicorn-utterances.com/posts/setup-standard-version

**Comment by JonnyBurger** [comment]
@khanzzirfan I'll try to make my commits more descriptive. However, in my opinion, conventional commits is too esoteric and I find myself I spend too much time on getting that right.

**Comment by JonnyBurger** [comment]
Will redo this feature cleanly

**Pull request closed without merging**